src/boolean.ts

```diff
@@ -0,0 +1,11 @@
+import { randomFromArray } from './fromArray.js'
+
+/**
+ * Returns either `true` or `false` with equal probability.
+ */
+export function randomBoolean(): boolean {
+
+  // randomFromArray returns undefined if the argument is an empty array.
+  // Since we know this argument is not empty, we cast it to get around the TS compiler.
+  return !!randomFromArray([true, false])
```

Comment:
I wonder if there's a better way to express this property of `randomFromArray`. Currently its definition looks like this:

```typescript
export function randomFromArray<T>(array: T[]): T | undefined
```

But it would be neat to say, the return type is `undefined` if `array` is empty, otherwise `T`. I played around a bit with a conditional type checking if `T extends never` but this didn't seem to work 🤔